src/utils_lua.c

```diff
@@ -274,38 +235,200 @@ int luaC_pushvalue(lua_State *L, value_t v, int ds_type) /* {{{ */
   return 0;
 } /* }}} int luaC_pushvalue */
 
-int luaC_pushvaluelist(lua_State *L, const data_set_t *ds,
-                       const value_list_t *vl) /* {{{ */
+int luaC_pushlabelset(lua_State *L, label_set_t label) /* {{{ */
 {
+  /*
+    label_set_t will be mapped to the following Lua table:
+
+    label = {
+      [1] => { name => value },
+      ...
+      [N] => { ... }
+    }
+   */
   lua_newtable(L);
+  if (label.num > 0) {
+    label_pair_t *ptr = label.ptr;
+    for (int i = 0; i < label.num; i++) {
+      lua_newtable(L);
+      lua_pushstring(L, ptr->value);
+      lua_setfield(L, -2, ptr->name);
+      /* label[N] = {} */
+      lua_rawseti(L, -2, i + 1);
+      ptr++;
+    }
+  }
+  return 0;
+} /* }}} int luaC_pushlabelset */
 
-  lua_pushstring(L, vl->host);
-  lua_setfield(L, -2, "host");
+int luaC_pushmetadata(lua_State *L, meta_data_t *meta) /* {{{ */
+{
+  /*
+    meta_data_t will be mapped to the following Lua table:
+
+    meta => {
+      [1] => {
+        type => ...,
```

Comment:
I'm not very familiar with Lua. From a quick look, it looks like Lua can determine the type at runtime. We we really need to export this field? If not, we could export the simpler `{ name0 => value0, name1 => value1, … }` table mentioned above.

Reply:
No need  to export for Lua world. I've removed. thanks!